lib/TopStoriesFeed.jsm

```diff
@@ -705,11 +705,9 @@ this.TopStoriesFeed = class TopStoriesFeed {
       return;
     }
     switch (action.type) {
-      // Check for pref initial values to lazy load activity stream top stories
-      // Here we are not using usual INIT and relying on PREFS_INITIAL_VALUES
-      // to check discoverystream pref and load activity stream top stories only if needed.
-      case at.PREFS_INITIAL_VALUES:
-        this.lazyLoadTopStories(action.data[DISCOVERY_STREAM_PREF]);
+      // Check discoverystream pref and load activity stream top stories only if needed.
+      case at.INIT:
+        this.lazyLoadTopStories(action.data[DISCOVERY_STREAM_PREF] || this.store.getState().Prefs.values[DISCOVERY_STREAM_PREF]);
```

Comment:
These should not be accessing action.data. INIT doesn't pass them. My suggestion for `getState()` is from `lazyLoadTopStories` side to avoid duplicating the code here and above.